### ISSUE TITLE
[Question] How to set maximumSize on DialogWindow/ComposeDialog? 

### ISSUE THREAD

**Issue by franmatesic** [JetBrains/compose-multiplatform#3830]
Setting maximumSize on ComposeDialog when creating a custom DialogWindow does not restrict resizing, but setting minimumSize does, as seen here:

![ezgif com-video-to-gif](https://github.com/JetBrains/compose-multiplatform/assets/25988237/6d14454e-c0e7-4bd5-8be8-7dd7fbd487b0)

[Custom dialog window implementation](https://gist.github.com/franmatesic/4b28c21a2e9b7e07240c6650e5bb40e0)

Creating custom dialog window:
```kotlin
TransparentDialog(
    resizable = true,
    maximumSize = Dimension(576, 1024),
    minimumSize = Dimension(360, 360),
) {
    // ...
}
``` 

I don't know if this is a Swing or Compose problem.

**Comment by MatkovIvan** [comment]
`maximumSize` here is `JDialog`s setter, it's not Compose.
Your snippet relays on other not-published code, so I had to simplify it to reproduce the issue on my end. So, yep Swing ignores max size if `isUndecorated == true`.
I tried to control it with this workaround:
```kt
addComponentListener(object : ComponentAdapter() {
    override fun componentResized(event: ComponentEvent?) {
        val width = size.width.coerceIn(minimumSize.width, maximumSize.width)
        val height = size.height.coerceIn(minimumSize.height, maximumSize.height)
        setSize(width, height)
    }
})
```
but it doesn't work well - the callback is called with delay causing visual glitches. I'm posting it here just to give you some direction for possible workarounds.

Anyway, I'm closing this topic as it's not related to Compose.
